lib/bitcoin/BitcoinProcessor.ts

```diff
@@ -528,14 +582,80 @@ export default class BitcoinProcessor {
       }
     }
 
+    if (this.isBatchBoundary(block)) {
+
+      // Compute the transaction fees for sampled transactions of this batch
+      const sampledTransactionIds = this.transactionSampler.getSample();
+      const sampledTransactionFees = new Array();
+      for (let transactionId of sampledTransactionIds) {
+        const transactionFee = await this.getTransactionFeeInSatoshi(transactionId);
+        sampledTransactionFees.push(transactionFee);
+      }
+
+      const batchId = this.getBatchId(block);
+      await this.quantileCalculator.add(batchId, sampledTransactionFees);
+
+      // Reset the sampler for the next batch
+      this.transactionSampler.clear();
+    }
+
     return blockHash;
   }
 
+  private isBatchBoundary (block: number): boolean {
+    return (block + 1) % this.proofOfFeeConfig.transactionFeeQuantileConfig.batchSizeInBlocks === 0;
+  }
+
+  private getBatchId (block: number): number {
+    return Math.floor(block / this.proofOfFeeConfig.transactionFeeQuantileConfig.batchSizeInBlocks);
+  }
+
+  /** Get the transaction out value in satoshi, for a specified output index */
+  private async getTransactionOutValueInSatoshi (txid: string, outIdx: number) {
+    const xact = await this.rpcCall({
+      method: 'getrawtransaction',
+      params: [
+        txid,  // transaction id
+        true   // verbose
+      ]
+    });
+
+    // output with the desired index
+    const vout = xact.vout.find((v: any) => v.n === outIdx);
+
+    return Math.round(vout.value * BitcoinProcessor.satoshiPerBTC);
```

Comment:
I remember fixing a bug sometime back - for some reason this value turned out to be non-integral - something like x.99999999, perhaps because satoshiperBTC is a really large number.